### PR TITLE
MNTOR-1809 - Add chart into dashboard top banner

### PR DESCRIPTION
**Pull request by codemist**
# References: 
Jira: MNTOR-1809

# Description

This adds the chart UI in the dashboard top banner. It also switches between two states:

This view before scanning: 
image

This view after scanning: 
image

Modal: 
image

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Comment by pdehaan** [comment]
Ah, and a general comment since I've been meaning to file this for a while&hellip;
Re: donut chart, we probably need a better variety of colors. Having 5 shades of very similar dark purples isn't easy to distinguish in a chart and for a long time I wasn't able to tell that the first two colors were different until I squinted and zoomed in on the chart.

**Comment by rhelmer** [comment]
> Ah, and a general comment since I've been meaning to file this for a while… Re: donut chart, we probably need a better variety of colors. Having 5 shades of very similar dark purples isn't easy to distinguish in a chart and for a long time I wasn't able to tell that the first two colors were different until I squinted and zoomed in on the chart.

I agree it's a bit hard to distinguish, if we change it we should be mindful of color insensitive vision (aka "color blindness"), there's an MDN article and explanation about how to use the Accessibility devtools: https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_Colors_and_Luminance

I took a quick look using the accessibility tools and the current chart was legible overall so I wouldn't want to make us worse for some folks while trying to make it better :)

**Review by rhelmer — APPROVED** [review]

File: locales-pending/premium.ftl [hunk 1]
```diff
@@ -38,6 +38,14 @@ exposure-chart-legend-heading-nr = Number
 # Variables:
 #   $nr (number) - Number of a particular type of exposure found for the user
 exposure-chart-legend-value-nr = { $nr }x
+exposure-chart-caption = This chart shows how many times your info is actively exposed.
+
+modal-active-number-of-exposures-title = About your number of active exposures
+# Variables:
+#   $limit (number) - Number of email addresses included in the plan
+modal-active-number-of-exposures-part-one = This chart includes the total number of times we found each type of data exposed across all data broker profiles and all data breaches for up to { $limit } email addresses that you are currently monitoring.
```
Comment (flodolo) [inline locales-pending/premium.ftl:46]: Please make this a plural string (see issue [here](https://github.com/mozilla/blurts-server/issues/3240#issuecomment-1645737650))

**Review by flodolo — APPROVED** [review]

File: src/app/(proper_react)/redesign/(authenticated)/user/dashboard/DashboardTopBanner.tsx [hunk 2]
```diff
@@ -15,33 +19,34 @@ export type DashboardTopBannerProps = {
...
     LetsFixDataContent: {
       headline: l10n.getString("dashboard-top-banner-protect-your-data-title"),
       description: l10n.getString(
         "dashboard-top-banner-protect-your-data-description",
         {
-          // TODO: Replace all mocked exposure data
-          data_breach_total_num: 95,
-          data_broker_total_num: 15,
+          data_breach_total_num: props.bannerData.dataBreachTotalNum,
+          data_broker_total_num: props.bannerData.dataBrokerTotalNum,
```
[the remark this answers is not in the document]
Comment (mansaj) [reply src/app/(proper_react)/redesign/(authenticated)/user/dashboard/DashboardTopBanner.tsx:38]: Resolved in: https://github.com/mozilla/blurts-server/pull/3252/files

File: src/app/components/client/Chart.tsx [hunk 3]
```diff
@@ -59,91 +74,141 @@ export const DoughnutChart = (props: Props) => {
...
+          {l10n.getString("exposure-chart-caption")}
+          <button
+            aria-label={l10n.getString("modal-open-alt")}
+            onClick={() => explainerDialogState.open()}
+          >
+            <QuestionMarkCircle
+              width="15"
+              height="15"
+              alt={l10n.getString("modal-open-alt")}
+            />
```
Comment (pdehaan) [inline src/app/components/client/Chart.tsx:193]: Not sure if it matters, but noticed sometimes the `<QuestionMarkCircle>` is 13&times;13, other times it's 15&times;15:

```sh
git grep -n -A3 "<QuestionMarkCircle" | grep -E "(QuestionMarkCircle|width=|height=)"

src/app/components/client/Chart.tsx:189:            <QuestionMarkCircle
src/app/components/client/Chart.tsx-190-              width="15"
src/app/components/client/Chart.tsx-191-              height="15"

src/app/components/client/ExposureCard.tsx:180:          icon={<QuestionMarkCircle alt="" width="13" height="13" />}

src/app/components/client/ExposureCard.tsx:223:            icon={<QuestionMarkCircle alt="" width="13" height="13" />}

src/app/components/client/ExposureCard.tsx:234:            icon={<QuestionMarkCircle alt="" width="13" height="13" />} // default icon for categories without a unique one

src/app/components/client/ExposuresFilter.tsx:238:              <QuestionMarkCircle
src/app/components/client/ExposuresFilter.tsx-239-                width="15"
src/app/components/client/ExposuresFilter.tsx-240-                height="15"

src/app/components/client/ExposuresFilter.tsx:254:              <QuestionMarkCircle
src/app/components/client/ExposuresFilter.tsx-255-                width="15"
src/app/components/client/ExposuresFilter.tsx-256-                height="15"

src/app/components/client/ProgressCard.tsx:119:          <QuestionMarkCircle alt="" width="15" height="15" />
```

File: src/app/components/client/Chart.tsx [hunk 4]
```diff
@@ -59,91 +74,141 @@ export const DoughnutChart = (props: Props) => {
...
-            .replace("</nr>", "")
-            .replace("<label>", "")
-            .replace("</label>", "")}
-          viewBox={`0 0 ${diameter} ${diameter}`}
-          className={styles.chart}
+  const modalContent = (
+    <div className={styles.modalBodyContent}>
+      <p>
+        {l10n.getString("modal-active-number-of-exposures-part-one", {
+          limit: 5,
```
Comment (pdehaan) [inline src/app/components/client/Chart.tsx:81]: Q: Is this magic `5` the same as `.env-dist:MAX_NUM_ADDRESSES=5`?

```sh
git grep "MAX_NUM_ADDRESSES" | cat

.env-dist:MAX_NUM_ADDRESSES=5
src/app/(nextjs_migration)/(authenticated)/user/breaches/page.tsx:              data-total={AppConstants.MAX_NUM_ADDRESSES}
src/app/(nextjs_migration)/(authenticated)/user/breaches/page.tsx:                    total: AppConstants.MAX_NUM_ADDRESSES,
src/app/(nextjs_migration)/(authenticated)/user/settings/page.tsx:                  limit: AppConstants.MAX_NUM_ADDRESSES,
src/app/(nextjs_migration)/(authenticated)/user/settings/page.tsx:                  Number.parseInt(AppConstants.MAX_NUM_ADDRESSES, 10)
src/app/(nextjs_migration)/components/server/AddEmailDialog.tsx:  const emailLimit = AppConstants.MAX_NUM_ADDRESSES;
src/app/api/v1/user/email/route.ts:      if (emailCount >= Number.parseInt(AppConstants.MAX_NUM_ADDRESSES, 10)) {
src/appConstants.js:  'MAX_NUM_ADDRESSES',
src/views/partials/breaches.js:    <figure class='email-stats' data-count=${data.emailTotalCount} data-total=${AppConstants.MAX_NUM_ADDRESSES}>
src/views/partials/breaches.js:        <strong>${getMessage('emails-monitored', { count: data.emailVerifiedCount, total: AppConstants.MAX_NUM_ADDRESSES })}</strong>
```

**Review by mansaj — APPROVED** [review]

File: locales/en/data-classes.ftl [hunk 5]
```diff
@@ -59,6 +59,7 @@ financial-investments = Financial investments
 financial-transactions = Financial transactions
 fitness-levels = Fitness levels
 flights-taken = Flights taken
+full-names = Full name
```
Comment (flodolo) [inline locales/en/data-classes.ftl:62]: @codemist 

I missed this one in review (not sure if it was there when I reviewed, but that's not relevant).

All categories in this file are plurals. Also, why is this category being added in this PR? Just a coincidence?

Comment (flodolo) [reply locales/en/data-classes.ftl:62]: Reusing strings is a bad idea, especially if we mix them from different sources. Some locales, for example, are translating categories lowercase.

Comment (flodolo) [reply locales/en/data-classes.ftl:62]: It's less than ideal in terms of localizability, but one way to minimize the impact would be to capitalize the first letter of the category (using a locale-sensitive library) when used in a different context.

**Pull request merged**